### DISCUSSION TITLE
Remix-auth adaptation to Next.js

### DISCUSSION THREAD

**Discussion by afoures** [sergiodxa/remix-auth#272] [General]
Hello,

I was wondering what would be the work to do to convert remix-auth to next js?

Thanks 😃

**Comment by sergiodxa** [comment]
You will need to change how the session works, since Remix Auth uses Remix's SessionStorage, or you can install `@remix-run/node` in your Next app and use Remix's SessionStorage since it doesn't depend on anything else from Remix.

Then you will need to stop throwing redirects since they won't be followed by Next, instead return the responses and in your code handle that. And provide your own `json` and `redirect` functions which are just wrappers of `new Response` anyway.

That's from Remix Auth side, how to use it inside Next.js and specially with App Router I have no idea, I imagine you would need to use it in a middleware since you can't redirect inside the components I believe, and then again in the RSC, and also integrate with those magic headers and cookies function, I don't know for sure.

I would recommend you to use another library built specifically for Next.js instead.
